src/_pytest/scope.py

```diff
@@ -0,0 +1,79 @@
+"""
+Scope definition and related utilities.
+
+Those are defined here, instead of in the 'fixtures', module because
+their use is spread across many other pytest modules, and centralizing it in 'fixtures'
+would cause circular references.
+
+Also this makes the module light to import, as it should.
+"""
+from enum import Enum
+from functools import lru_cache
+from typing import Optional
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing_extensions import Literal
+
+    _ScopeName = Literal["session", "package", "module", "class", "function"]
+
+
+class Scope(Enum):
```

Comment:
i just looked up the details of enum

lets use the example of https://docs.python.org/3/library/enum.html#when-to-use-new-vs-init
and store the name and the index togethe so its
```
 Session = ("session", 1)
 Package = ("package", 2)
```

going overboard we might want to use `auto()` + its heleprs to get al lthe things in

for future reference we might want to add the definition scope one before function (its intended to eventually come in between the collectors and the functions to contain parameterization)

Reply:
Why do you feel having the index there is important? See the comment from @bluetech:

https://github.com/pytest-dev/pytest/pull/8913#pullrequestreview-708086309

I was going to try to get rid of that index altogether. Do you feel it is necessary to have it there for other reasons than sorting them?